clang/include/clang/Basic/BuiltinsRISCV.td

```diff
@@ -20,6 +20,12 @@ class RISCVBuiltin<string prototype, string features = ""> : TargetBuiltin {
 
 let Attributes = [NoThrow, Const] in {
 //===----------------------------------------------------------------------===//
+// Zicsr extension.
+//===----------------------------------------------------------------------===//
+def csrr : RISCVBuiltin<"unsigned long int(unsigned long int)", "zicsr">;
```

Comment:
If you want "size_t" or "uintptr_t", please use that, not "unsigned long".

Reply:
I don't think these are `Const` since they operate on state that is not part of their arguments/return.

Reply:
I can certainly change it to `size_t`. And I'll move them out of the section marked with `Const`.